### ISSUE TITLE
Head breaks apps that use widgets with type parameter

### ISSUE THREAD

**Issue by hsparra** [bokeh/bokeh#4101]
After the Typescript merge any application or bokeh code that uses a widget with the **type** parameter, such as a Button, will error with an AttributeError.

Recommend: Use a deprecation warning instead or "magically" convert it behind the scenes when running so it does not error.

To Recreate: Run any app or Bokeh code with a widget that contains the type parameter that run under 0.11.1. Alternatively, run the following Bokeh code (taken from Bokeh docs) against head:

```
from bokeh.models.widgets import Button
from bokeh.io import output_file, show, vform

output_file("button.html")

button = Button(label="Foo", type="success")

show(vform(button))
```

**Comment by bryevdv** [comment]
ping @mattpap need a deprecation here

**Comment by mattpap** [comment]
@bryevdv, fixed in PR #4107.
